### PR TITLE
[extractor/kompas] Add extractor

### PR DESCRIPTION
**Pull request by HobbyistDev**
### Description of your *pull request* and other information

</details>

ADD DESCRIPTION HERE
This PR add site-support for video.kompas.com. Kompas is an online news provider in Indonesia. Kompas.com seem to use jixie player (player that based in singapure) as video player. This player actually provide drm video but we can found non-drm video at api call

Fixes #

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

File: yt_dlp/extractor/kompas.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        return {
+            'id': video_id,
+            'display_id': display_id,
+            'formats': formats,
+            'subtitles': subtitles,
+            'title': json_data.get('title') or self._html_search_meta(['og:title', 'twitter:title'], webpage),
+            'description': (clean_html(traverse_obj(json_data, ('metadata', 'description')))
+                            or self._html_search_meta(['description', 'og:description', 'twitter:description'], webpage)),
+            'thumbnails': traverse_obj(json_data, ('metadata', 'thumbnails')),
+            'thumbnail': traverse_obj(json_data, ('metadata', 'thumbnail')),
```
Comment (pukkandan) [inline yt_dlp/extractor/kompas.py:64]: return only one of these

File: yt_dlp/extractor/kompas.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            'duration': float_or_none(traverse_obj(json_data, ('metadata', 'duration'))),
+            'has_drm': json_data.get('drm'),
```
Comment (pukkandan) [inline yt_dlp/extractor/kompas.py:66]: this should be set inside the format

File: yt_dlp/extractor/kompas.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            'tags': str_or_none(traverse_obj(json_data, ('metadata', 'keywords')), '').split(',') or None,
+            'categories': str_or_none(traverse_obj(json_data, ('metadata', 'categories')), '').split(',') or None,
```
Comment (pukkandan) [inline yt_dlp/extractor/kompas.py:68]: `''.split(',') or None` doesn't work like you think. It makes sense to use `try_call` here instead
```suggestion
            'tags': try_call(lambda: json_data['metadata']['keywords'].split(',')),
            'categories': try_call(lambda: json_data['metadata']['categories'].split(',')),
```

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/kompas.py (hunk 1 above)
Comment (HobbyistDev) [reply yt_dlp/extractor/kompas.py]: Resolved in https://github.com/yt-dlp/yt-dlp/pull/4562/commits/638c0471c43a42d976e61ac5c474757e1fb11aef

File: yt_dlp/extractor/kompas.py (hunk 2 above)
Comment (HobbyistDev) [reply yt_dlp/extractor/kompas.py:66]: Can i just remove this metadata field ? i don't have any link from api that actually has drm

Comment (pukkandan) [reply yt_dlp/extractor/kompas.py:65]: sure

**Pull request merged**